lib/rouge/lexers/solidity.rb

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Solidity < RegexLexer
+      title "Solidity"
+      desc "Solidity, an Ethereum smart contract programming language"
+      tag 'solidity'
+      filenames '*.sol', '*.solidity'
+      mimetypes 'text/x-solidity'
+
+      # optional comment or whitespace
+      ws = %r((?:\s|//.*?\n|/[*].*?[*]/)+)
+      id = /[a-zA-Z$_][\w$_]*/
+
+      def self.detect?(text)
+        return true if text.start_with? 'pragma solidity'
+      end
+
+      # TODO: seperate by "type"
+      def self.keywords
+        @keywords ||= Set.new %w(
+          abstract anonymous as assembly break catch calldata constant
+          constructor continue contract do delete else emit enum event
+          external fallback for function hex if indexed interface
+          internal import is library mapping memory modifier new
+          override payable public pure pragma private receive return
+          returns storage struct throw try type using var view virtual
+          while
+        )
+      end
+
+      def self.builtins
+        return @builtins if @builtins
+
+        @builtins = Set.new %w(
+          now
+          false true
+          balance now selector super this
+          blockhash gasleft
+          assert require revert
+          selfdestruct suicide
+          call callcode delegatecall
+          send transfer
+          addmod ecrecover keccak256 mulmod sha256 sha3 ripemd160
+        )
+
+        # TODO: use (currently shadowed by catch-all in :statements)
+        abi = %w(encode encodePacked encodeWithSelector encodeWithSignature)
+        @builtins.merge( abi.map { |i| "abi.#{i}" } )
+        block = %w(coinbase difficulty gaslimit hash number timestamp)
+        @builtins.merge( block.map { |i| "block.#{i}" } )
+        msg = %w(data gas sender sig value)
+        @builtins.merge( msg.map { |i| "msg.#{i}" } )
+        tx = %w(gasprice origin)
+        @builtins.merge( tx.map { |i| "tx.#{i}" } )
+      end
+
+      def self.constants
+        @constants ||= Set.new %w(
+          wei finney szabo ether
+          seconds minutes hours days weeks years
+        )
+      end
+
+      def self.keywords_type
+        return @keywords_type if @keywords_type
+
+        @keywords_type = Set.new %w(
+          address bool byte bytes int string uint
+        )
+
+        # bytes1 .. bytes32
+        @keywords_type.merge( (1..32).map { |i| "bytes#{i}" } )
+
+        # size helper
+        sizes = (8..256).step(8)
+        # [u]int8 .. [u]int256
+        @keywords_type.merge( sizes.map { |n|  "int#{n}" } )
+        @keywords_type.merge( sizes.map { |n| "uint#{n}" } )
```

Comment:
This won't generate as many elements as was the case with the `fixed` / `ufixed` keywords, but can this similarly be simplified in the rule by matching any digit?